src/gui/owncloudgui.cpp

```diff
@@ -56,16 +56,40 @@ namespace OCC {
 
 const char propertyAccountC[] = "oc_account";
 
+ownCloudGui *ownCloudGui::_instance = nullptr;
+
+ownCloudGui *ownCloudGui::instance(Application *parent)
+{
+    if (!_instance) {
+        Q_ASSERT(parent);
+        _instance = new ownCloudGui(parent);
+    }
+    return _instance;
+}
+
 ownCloudGui::ownCloudGui(Application *parent)
     : QObject(parent)
     , _tray(nullptr)
-    , _settingsDialog(new SettingsDialog(this))
+    , _settingsDialog(nullptr)
     , _logBrowser(nullptr)
 #ifdef WITH_LIBCLOUDPROVIDERS
     , _bus(QDBusConnection::sessionBus())
 #endif
     , _app(parent)
 {
+    Q_ASSERT(!_instance);
+    _instance = this;
+}
+
+ownCloudGui::~ownCloudGui()
+{
+    _instance = nullptr;
+}
+
+void ownCloudGui::init()
+{
+    _settingsDialog = new SettingsDialog(this);
```

Comment:
Hm... actually, can't we go one step further and create it only when we try to show it for the first time?

Reply:
Actually this is already implemented in `ownCloudGui::slotShowSettings`

Reply:
I've tried this approach but it caused a crash in `SettingsDialog::customizeStyle`. However, I fixed the crash before finishing this comment. Since it has some other implications, let's handle it in a dedicated PR :-)